### PR TITLE
Adds `handshake` icon

### LINKED ISSUE #608: [Icon Requests from Feather] `handshake`
As per https://github.com/lucide-icons/lucide/issues/119 and https://github.com/feathericons/feather/issues/444 and https://github.com/feathericons/feather/issues/486:

image
handshake

### PR DESCRIPTION
**Pull request by karsa-mistmere**
closes #608 

Also see: https://github.com/lucide-icons/lucide/issues/119 and https://github.com/feathericons/feather/issues/444 and https://github.com/feathericons/feather/issues/486

## What is the purpose of this pull request?
- [x] New Icon

### Description
Adds handshake icon to represent businesses, deals, partnerships etc.

### Icon use case 
As above.

### Alternative icon designs
See #608 

### Concept
- [x] I have provided valid use cases for each icon.
- [x] I have not added any a brand or logo icon.
- [x] I have not used any hate symbols.
- [x] I have not included any religious or political imagery.

### Author, credits & license

- [x] The icons are solely my own creation.

### Naming 

- [x] I've read and followed the [naming conventions](https://lucide.dev/guide/design/icon-design-guide#naming-conventions)
- [x] I've named icons by what they are rather than their use case.
- [x] I've provided meta JSON files in `icons/[iconName].json`.

### Design 

- [x] I've read and followed the [icon design guidelines](https://lucide.dev/guide/design/icon-design-guide)
- [x] I've made sure that the icons look sharp on low DPI displays.
- [x] I've made sure that the icons look consistent with the icon set in size, optical volume and density.
- [x] I've made sure that the icons are visually centered.
- [x] I've correctly optimized all icons to two points of precision.

## Before Submitting 

- [x] I've read the [Contribution Guidelines](https://github.com/lucide-icons/lucide/blob/main/CONTRIBUTING.md).
- [x] I've checked if there was an existing PR that solves the same issue.

### CODE REVIEW

**Comment by jguddas** [comment]
I'm not a fan, I feel like it's very hard to identify.

**Comment by karsa-mistmere** [comment]
> I'm not a fan, I feel like it's very hard to identify.

I made this alternate design but it feels a bit clunky:
image

**Comment by jguddas** [comment]
> > I'm not a fan, I feel like it's very hard to identify.
> 
> I made this alternate design but it feels a bit clunky: image

Feels better, but yes a bit clunky.

**Comment by ericfennis** [comment]
@karsa-mistmere Can you post a 100% scale version? 
The current `hand` also looks a bit clunky when it is larger than 100%

**Comment by karsa-mistmere** [comment]
image

**Comment by ericfennis** [comment]
@karsa-mistmere Hmm looks not "bad", it's quite recognizable in comparison with the icon in the PR. 
But not perfect yet, can you maybe send the SVG? I want to give it a try.

**Comment by ericfennis** [comment]
@karsa-mistmere @jguddas I've adjusted a bit (left from karse, second one adjusted by me). 
I've reduced the length of the fingers a bit. And created a curve in the thumb like the `hearth-hands` icon
To me, this variant of the `handshake` is more in line with our other hand icons.

image

100%:
handshake

**Comment by karsa-mistmere** [comment]
> @karsa-mistmere @jguddas I've adjusted a bit (left from karse, second one adjusted by me). I've reduced the length of the fingers a bit. And created a curve in the thumb like the `hearth-hands` icon To me, this variant of the `handshake` is more in line with our other hand icons.
> 
> image
> 100%: handshake

I really like this one, especially how it works together with `heart-handshake`, can you update the PR with it?

**Comment by ericfennis** [comment]
@karsa-mistmere Updated!

**Comment by karsa-mistmere** [comment]
Should we also update `heart-handshake` to match this orientation? 🤔

image

**Comment by ericfennis** [comment]
Hmm, I was looking at what [Google](https://www.google.com/search?q=handshake+icon&oq=handshake+icon&sourceid=chrome&ie=UTF-8) gives me back as a `handshake` icon.
I suggest mirroring the `handshake` icon to be more conventional

**Comment by karsa-mistmere** [comment]
> Hmm, I was looking at what [Google](https://www.google.com/search?q=handshake+icon&oq=handshake+icon&sourceid=chrome&ie=UTF-8) gives me back as a `handshake` icon. I suggest mirroring the `handshake` icon to be more conventional

Done!

File: icons/handshake.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+  "$schema": "../icon.schema.json",
+  "contributors": [
+    "karsa-mistmere"
```
Comment (jguddas) [inline icons/handshake.json:4]:
```suggestion
    "karsa-mistmere",
    "ericfennis"
```

**Review by jguddas — CHANGES_REQUESTED** [review]

**Comment by jguddas** [comment]
What about updating the handshake heart to use the same round fingers?

icon (37) 1

**Comment by karsa-mistmere** [comment]
> What about updating the handshake heart to use the same round fingers?
> 
> icon (37) 1

I've tried something similar, but does this not make it look like a regular (albeit rounded) handshake as opposed to a heart as intended?

**Review by ericfennis — APPROVED** [review]

**Pull request merged**